modules/dhcp/dhcp_api.rb

```diff
@@ -66,6 +69,7 @@ class Proxy::DhcpApi < ::Sinatra::Base
 
   # returns a record for a mac address
   get "/:network/mac/:mac_address" do
+    server.validate_supported_address(params[:ip])
```

Comment:
Ditto, pretty much every route (method) in this file has an incorrect. Sinatra uses parameter keys as defined in the argument of the HTTP method: http://sinatrarb.com/intro.html